release/models/hashing/openconfig-hashing.yang

```diff
@@ -0,0 +1,238 @@
+                    hash.";
+            }
+            enum TARGET_DEFINED {
+                description "Use the default fields that are used by
+                    the device for calculating the hash.";
+            }
+            enum FLOW_LABEL {
+                description "Use the flow label in the IPv6 header
+                    to calculate the hash.";
+            }
+        }
+        description "The IPv6 fields that should be used to
+            compute the hash.";
+    }
+
+    typedef hash-field-type-transport {
+        type enumeration {
+            enum NONE {
+                description "Do not use any transport fields in the
+                    calculation of the hash.";
+            }
+            enum SRC_PORT {
+                description "Use only the source port from the
+                    transport header in the calculation of the hash.";
+            }
+            enum DST_PORT {
+                description "Use only the destination port from the
+                    transport header in the calculation of the hash.";
+            }
+            enum BOTH {
+                description "Use both the source and
+                    destination ports from the transport header in
+                    the calculation of the hash.";
+            }
+            enum TARGET_DEFINED {
+                description "Use the default fields that are used by
+                    the device for calculating the hash.";
+            }
+        }
+        description "The trasport layer fields that should be used to
+            compute the hash.";
+    }
+
+    grouping hashing-inputs {
+        description
+            "Top level container for inputs to be used for hashing policies.";
+
+        leaf ipv4 {
+            type hash-field-type-ipv4;
+            description "The IPv4 fields that should be used to
+                compute the hash.";
+        }
+        leaf ipv6 {
+            type hash-field-type-ipv6;
+            description "The IPv6 fields that should be used to
+                compute the hash.";
+        }
+        leaf transport-ports {
+            type hash-field-type-transport;
+            description "The transport layer fields that should be used to
+                compute the hash.";
+        }
+    }
+
+    grouping hashing-policy-config {
+        description
+          "Configuration data for hashing policies.";
+
+        leaf name {
+            type string;
+            description
+              "The name of the hashing policy.
+
+              When a configured user-controlled policy is created by the
+              system, it is instantiated with the same name in the
+              /policies/policy[name]/state list.";
+        }
+
+        leaf algorithm {
+            type string;
+            description
+              "The algorithm used when calculating a hash based on packet headers";
+        }
+
+        leaf offset {
```

Comment:
Can we cite where these are actually configurable? the references in the PR comments don't seem to show this is something that is generally supported.

In the case that we are looking to add extensions that are for specific types of devices that might expose this -- or that this is something that is to be supported in the future, please look at the pattern of using a `-ext` module.

Reply:
this leaf is no longer present in #959